### ISSUE TITLE
hide image

### ISSUE THREAD

**Issue by dvs1** [qutebrowser/qutebrowser#2958]
Would it be possible to add "hide image" to the context menu of images?
I don't need it to be permanent.

The only thing I could find is in browser/webkit/webview.py:
```
    def contextMenuEvent(self, e):
        """Save a reference to the context menu so we can close it.
        This is not needed for QtWebEngine, so it's in here.
        """
        menu = self.page().createStandardContextMenu()
        self.shutting_down.connect(menu.close)
        modeman.instance(self.win_id).entered.connect(menu.close)
        menu.exec_(e.globalPos())
```

And from https://doc.qt.io/qt-5/qwebenginecontextmenudata.html

> The QWebEngineContextMenuData class provides context data for populating or extending a context menu with actions.

Does this mean that QWebEngine is automatically producing the context menu we currently see?

If I was going to attempt to extend this menu, 
is browser/webengine/webview.py the correct location?

🍀

**Comment by The-Compiler** [comment]
See #349 - indeed, right now qutebrowser does nothing related to the context menu basically. To customize it, most likely all its functionality needs to be recreated (at least for QtWebKit, for QtWebEngine `QWebEnginePage::createStandardContextMenu()` might help). It's probably not a trivial task to get working with both backends, unfortunately.

**Comment by dvs1** [comment]
Does closing this issue mean you would reject "hide image" as a standard feature in any form?

**Comment by NoctuaNivalis** [comment]
I don't think it was, just that you can't do it with a context menu.

However, I think this is already possible with the current codebase. You could bind a key to hint images, and call some javascript with the selected image URL to hide the actual image.

**Comment by dvs1** [comment]
Thanks.

It's still not clear why the issue was closed.

Context menu customization seems to be a desired feature.
Once that feature has been implemented it then becomes possible to add "hide image".

The title of this issue allows for multiple implementations.

**Comment by The-Compiler** [comment]
It was closed because #349 already covers "there should be a customizable context menu".

I don't think this is a common enough thing people want to warrant a default entry in the context menu, but with the new config and #349, it might get as easy as something like `ui.context_menu['Hide image'] = 'js this.removeFromDocument();'` or so.
